storage/rocksdb/rdb_compact_filter.h

```diff
@@ -65,6 +65,17 @@ class Rdb_compact_filter : public rocksdb::CompactionFilter {
 
       if (!m_should_delete) {
         get_ttl_duration(gl_index_id, &m_ttl_duration);
+
+        /*
+          For efficiency reasons, we lazily call GetIntProperty to get the
+          oldest snapshot time, each time we find a new index to be processed.
+        */
+        rocksdb::DB *const rdb = rdb_get_rocksdb_db();
```

Comment:
Regarding the check to `m_snapshot_timestamp > 0`.  I was thinking to purposely check for a 'newer' snapshot between every index change, not once per compaction.  

Are you suggesting that we use the same `m_snapshot_timestamp` for the duration of the entire compaction? (I'm fine with this, just making sure that this is what you meant)